reference/pipeline/expressions/index.md

```diff
@@ -0,0 +1,230 @@
+
+Another example is that the status attribute of a stage is actually an enum
+internally, not a string. To compare the status to a string, you need to call
+`.toString()` on the result. For example:
+`${#stage("Deploy")["status"].toString() == "SUCCEEDED"}`.
+
+### Functions
+
+The expression language has several built in helper functions that simplify
+common use cases. The syntax for calling built-in functions is
+`#functionName(params)`. For example, `#fromUrl("http://www.netflix.com")`. See
+the full list of functions [here](#helper-functions).
+
+### Lists
+
+You can also use lists in your expressions. Spinnaker always provides a list
+called _stages_, which you can use to access each stage by index:
+`${execution["stages"][0]}` returns the value of the first stage in your
+pipeline.
+
+Note that `stages` is mostly referenced here for illustration purposes. The
+value of `stages[i]` depends on the order in which your stages execute, which
+makes it fragile. The recommended way to access a specific stage is to use the
+[`#stage("Stage Name")` helper function](#stagestring).
+
+### Maps
+
+You can use maps to access data from the JSON representation of your pipeline.
+For example, `${trigger["properties"]["example"]}` evaluates to the value of
+the `example` trigger property. To see the available values for a given
+pipeline, view the execution JSON:
+1. Go to an execution of your pipeline.
+2. Click on _Details_ to expand the pipeline.
+3. Click on the source link under the pipeline execution, which opens a new tab
+    containing the JSON details of your pipeline execution.
+    _[[screenshot](https://screenshot.googleplex.com/U1rVRjdiqKC.png)]_
+
+Note that while Spinnaker supports both dot (`map.value`) and square bracket
+(`map["value"]`) notation, we recommend using square brackets. That is, prefer
+`trigger["properties"]["value"]` instead of `trigger.properties.value`. There
+are a few places where using dot notation produces unexpected results, such as
+after a filter operation or when getting nested JSON values from an URL. Bracket
+notation also allows you to access non-alphanumeric properties.
+
+You can filter maps using `.?`. For example, to filter a list of stages by type,
+you would use `${execution["stages"].?[type == "bake"]}`. The result is a
+list, and can be accessed as such: `${execution["stages"].?[type == "bake"][0]}`
+returns the first bake stage in your pipeline.
+
+### Math
+
+You can use arithmetic operations in your expressions, such as
+`${trigger["buildInfo"]["number"] * 2}`.
+
+### Strings
+
+Strings evaluate to themselves: `${"This is a String."}` becomes "This is a
+String."
+
+## Helper properties
+
+[Helper properties](/guides/user/pipeline/expressions/#helper-properties) are
+attribute shortcuts that you can use in Spinnaker. The specific properties are:
+
+* `execution`: refers to the current pipeline execution.
+* `parameters`: refers to pipeline parameters.
```

Comment:
This is an alias to trigger.parameters. it might make sense to reference trigger.parameters more explictly in the docs